sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+    - You need to customize the polling strategy
+ 2) [Polling method - LROBasePolling/AsyncLROBasePolling](#polling-method---lrobasepollingasynclrobasepolling)
+    - You need to customize the polling loop
+ 3) [Poller - LROPoller/AsyncLROPoller](#poller---lropollerasynclropoller)
+    - You need to customize the public interface of the poller
+
+
+### LRO algorithm/strategies - OperationResourcePolling, LocationPolling, StatusCheckPolling
+
+`azure.core.polling` provides three built-in strategies for polling - `OperationResourcePolling`, `LocationPolling`, and `StatusCheckPolling`.
+The type of polling needed will be determined automatically, unless otherwise changed by the client library developer. If the LRO is determined
+not to fit `OperationResourcePolling` or `LocationPolling`, `StatusCheckPolling` serves as a fallback strategy which will not perform polling, but
+instead return a successful response for a 2xx.
+
+If you need to customize the polling strategy, choose a polling algorithm that closely represents what you need to do
+or create your own that inherits from `azure.core.polling.LongRunningOperation` and implements the necessary methods.
+
+For our purposes, let's say that `OperationResourcePolling` closely resembles what the service does, but we
+need to account for a non-standard status.
+
+Example: Raise exception for a non-standard status that is returned via the initial POST call - "ValidationFailed".
+
+```python
+class CustomOperationResourcePolling(OperationResourcePolling):
+    """Implements a operation resource polling, typically from Operation-Location. Customized to raise an
+    exception in the case of a "ValidationFailed" status returned from the service.
+
+    :param str operation_location_header: Name of the header to return operation format (default 'operation-location')
+    :keyword dict[str, any] lro_options: Additional options for LRO. For more information, see
+     https://aka.ms/azsdk/autorest/openapi/lro-options
+    """
+
+    def get_status(self, pipeline_response: "PipelineResponseType") -> str:
+        response = pipeline_response.http_response
+        if _is_empty(response):
+            raise BadResponse("The response from long running operation does not contain a body.")
+
+        body = _as_json(response)
+        status = body.get("status")
+        if not status:
+            raise BadResponse("No status found in body")
+        if status.lower() == "validationfailed":
+            raise HttpResponseError(response=response, error=body["error"])
+        return status
+```
+
+To plug into a client method which was generated as an LRO, you can pass the `polling` keyword argument. The `polling`
+keyword argument takes a `azure.core.polling.PollingMethod` and allows for a custom strategy to be passed in to the keyword
```

Comment:
The snippet is demonstrating `LROBasePolling` rather than `PollingMethod` - so we might want to reconcile that, either by mentioning it here, or a comment in the snippet.

Reply:
Ah, I see you explain that the bottom. Maybe then we could just update this line to make it clearer that we are referring here to a the PollingMethod protocol - maybe something like:
```suggestion
keyword argument takes an implementation of `azure.core.polling.PollingMethod` (e.g. `azure.core.polling.LROBasePolling`) and allows for a custom strategy to be passed in to the keyword
```
Could we also make `azure.core.polling.PollingMethod` a link to the reference docs?